examples/ArduinoWiFi101/ArduinoWiFi101.ino

```diff
@@ -62,10 +66,6 @@ void loop() {
   }
 }
 
-void messageReceived(String topic, String payload, char * bytes, unsigned int length) {
-  Serial.print("incoming: ");
-  Serial.print(topic);
-  Serial.print(" - ");
-  Serial.print(payload);
-  Serial.println();
+void messageReceived(String topic, String payload) {
```

Comment:
@256dpi your solution  (just use Strings with binary data) sounds like the best way to do it given the current situation -- has anyone hit a problem where their binary data had a 0 in the middle?  That's the only issue I could foresee. It definitely highlights the value of something like `Buffer`. 

In that context, the const String idea makes sense. 

There's no method in the library for separating the incoming topic, is there? From the little I've seen, it seems like `device/property` is pretty common in a topic, no? Might be worthwhile to make a helper function that lets you pull the two apart, something like this:
```

 // parse the topic
  int divider = topic.indexOf('/');
  String deviceName = topic.substring(0, divider);
  String property = topic.substring(divider + 1);

```
I guess you could do it with a couple of getters. Or just make it standard practice in an example or two.